core/src/main/scala/org/apache/spark/storage/BlockManagerId.scala

```diff
@@ -132,10 +133,17 @@ private[spark] object BlockManagerId {
     getCachedBlockManagerId(obj)
   }
 
-  val blockManagerIdCache = new ConcurrentHashMap[BlockManagerId, BlockManagerId]()
+  /**
+   * Here we set max cache size as 10000.Since the size of a BlockManagerId object
```

Comment:
nit:
```
The max cache size is hardcoded to 10000, since the size of a BlockManagerId object is about 48B, the total memory cost should be below 1MB which is feasible.
```